src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -1199,29 +1199,28 @@ private async Task<CompilationCompletedEvent> ProcessCompilationEventsCoreAsync(
                     }
 
                     CompilationEvent e;
-                    try
+                    if (!CompilationEventQueue.TryDequeue(out e))
                     {
-                        if (!CompilationEventQueue.TryDequeue(out e))
+                        if (!prePopulatedEventQueue)
                         {
-                            if (!prePopulatedEventQueue)
-                            {
-                                e = await CompilationEventQueue.DequeueAsync(cancellationToken).ConfigureAwait(false);
-                            }
-                            else
+                            var optionalEvent = await CompilationEventQueue.TryDequeueAsync(cancellationToken).ConfigureAwait(false);
+                            if (!optionalEvent.HasValue)
                             {
-                                return completedEvent;
+                                // When the queue is completed with a pending TryDequeueAsync return then
```

Comment:
>When the queue is completed with a pending TryDequeueAsync return then [](start = 35, length = 70)

I find this wording confusing, consider rewording.

Reply:
I amended the previous commit just to reword this. The difference can be seen in this:
https://github.com/dotnet/roslyn/compare/561f07655fc1c829a0948546691867533051739c..2043e01430be991df037cb94d9f1b4b0fd8713e7